hack/run-ci-e2e-test.sh

```diff
@@ -46,15 +45,58 @@ NODE_COUNT=${NODE_COUNT:-2}
 SKIP_NODE_DELETION=${SKIP_NODE_DELETION:-"-skip-node-deletion=false"}
 KEY_PAIR_NAME=${KEY_PAIR_NAME:-"libra"}
 
+# OPERATOR_IMAGE defines where the WMCO image to test with is located. If $OPERATOR_IMAGE is already set, use its value.
+# Setting $OPERATOR_IMAGE is required for local testing.
+# In OpenShift CI $IMAGE_FORMAT will be set to a value such as registry.svc.ci.openshift.org/ci-op-<input-hash>/stable:${component}
+# We need to replace ${component} with the name of the WMCO image.
+# The stable namespace is used as it will use the last promoted version when the test is run in an repository other than this one.
+# When running in this repository the image built in the pipeline will be used instead.
+# https://steps.svc.ci.openshift.org/help/ci-operator#release
+OPERATOR_IMAGE=${OPERATOR_IMAGE:-${IMAGE_FORMAT//\/stable:\$\{component\}//stable:windows-machine-config-operator-test}}
+
+# Create a temporary directory to hold the edited manifests which will be removed on exit
+MANIFEST_LOC=`mktemp -d`
+trap "rm -r $MANIFEST_LOC" EXIT
+cp -r deploy/olm-catalog/windows-machine-config-operator/ $MANIFEST_LOC
+sed -i "s|REPLACE_IMAGE|$OPERATOR_IMAGE|g" $MANIFEST_LOC/windows-machine-config-operator/manifests/windows-machine-config-operator.clusterserviceversion.yaml
+
 cd $WMCO_ROOT
 oc create -f deploy/namespace.yaml
+oc create secret generic cloud-credentials --from-file=credentials=$AWS_SHARED_CREDENTIALS_FILE -n windows-machine-config-operator
+oc create secret generic cloud-private-key --from-file=private-key.pem=$KUBE_SSH_KEY_PATH -n windows-machine-config-operator
+oc create secret generic kubeconfig --from-file=kubeconfig=$KUBECONFIG -n windows-machine-config-operator
+
+# Run the operator in the windows-machine-config-operator namespace
+# Currently this fails even on successes, adding this check to ignore the failure
+# https://github.com/operator-framework/operator-sdk/issues/2938
+if ! $OSDK run --olm --olm-namespace openshift-operator-lifecycle-manager --operator-namespace windows-machine-config-operator --operator-version 0.0.0 --manifests $MANIFEST_LOC/windows-machine-config-operator; then
+  echo operator-sdk run failed
+fi
+
 # The bool flags in golang does not respect key value pattern. They follow -flag=x pattern.
 # -flag x is allowed for non-boolean flags only(https://golang.org/pkg/flag/)
 # Run the creation tests and skip deletion of the Windows VMs
-$OSDK test local ./test/e2e --debug --up-local --operator-namespace=windows-machine-config-operator --local-operator-flags "--zap-level=debug --zap-encoder=console" --go-test-flags "-run=TestWMCO/create -v -timeout=60m -node-count=$NODE_COUNT -skip-node-deletion -ssh-key-pair=$KEY_PAIR_NAME"
+if ! $OSDK test local ./test/e2e --no-setup --debug --operator-namespace=windows-machine-config-operator --go-test-flags "-run=TestWMCO/create -v -timeout=60m -node-count=$NODE_COUNT -skip-node-deletion -ssh-key-pair=$KEY_PAIR_NAME"; then
+  get_wmco_logs
+  exit 1
+fi
```

Comment:
I still think these can be modularized using [bash functions that take inputs](https://linuxize.com/post/bash-functions/#passing-arguments-to-bash-functions). Inputs would be the tests to run and based on that you can construct the arguments.

Reply:
This has not been addressed.